src/ui/RealmBrowser/index.tsx

```diff
@@ -295,15 +298,62 @@ class RealmBrowserContainer
     const mightBeEncrypted =
       message.indexOf('Not a Realm file.') >= 0 ||
       message.indexOf('Invalid mnemonic') >= 0;
+    const realm = this.props.realm;
     if (mightBeEncrypted) {
       this.setState({
         isEncryptionDialogVisible: true,
         progress: {
           status: 'done',
         },
       });
+    } else if (
+      message === FILE_UPGRADE_NEEDED_MESSAGE &&
+      realm.mode === realms.RealmLoadingMode.Local
+    ) {
+      const buttons = ['Cancel', 'Upgrade in-place', 'Backup and upgrade'];
+      const answerIndex = remote.dialog.showMessageBox({
+        type: 'question',
+        buttons,
+        defaultId: 2,
+        title: 'Realm file needs an upgrade',
+        message: 'The Realm file stores data in an outdated format',
+        detail:
+          'This file needs to be upgraded to a newer file format before it can be opened. Would you like a backup of the file, before performing an irreversible upgrade of the file?',
```

Comment:
```suggestion
          'This file needs to be upgraded to a newer file format before it can be opened by this version of Studio. Would you like a backup of the file, before performing an irreversible upgrade of the file?',
```

Reply:
Not sure if we should inform about the option to simply open the file in an older version of Studio. Or is that too obvious?

Reply:
We don't necessarily release compatibility versions for every version of the file format - and it's not trivial for a user to determine which compatibility version they should download (or where they are downloaded from). I think we can safely assume that the vast majority of users wouldn't be able to download a version which is compatible with a particular the file format version.

I suggest not adding " by this version of Studio." as the message is already somewhat long and a slightly complicated to understand.